src/main/java/service/CryptoCurrencyService.java

```diff
@@ -0,0 +1,58 @@
+package service;
+
+import entity.CryptoCurrency;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import repository.CryptoCurrencyRepository;
+
+import javax.persistence.EntityNotFoundException;
+import java.math.BigDecimal;
+import java.util.List;
+
+@Service
+public class CryptoCurrencyService {
+
+    private final CryptoCurrencyRepository cryptoCurrencyRepository;
+
+    @Autowired
+    public CryptoCurrencyService(CryptoCurrencyRepository cryptoCurrencyRepository) {
+        this.cryptoCurrencyRepository = cryptoCurrencyRepository;
+    }
+
+    public CryptoCurrency createCryptoCurrency(String currencyCode, String name, BigDecimal usdRate) {
+        CryptoCurrency existingCryptoCurrency = cryptoCurrencyRepository.findByCurrencyCode(currencyCode).orElse(null);
```

Comment:
I would suggest creating a method in a repository to check whether it exists by currency code